superset-frontend/src/cccs-viz/plugins/plugin-chart-at-a-glance-user-id/src/AtAGlanceUserID.tsx

```diff
@@ -0,0 +1,268 @@
+import React, { useState } from 'react';
+import { RiGlobalFill } from 'react-icons/ri';
+import { QueryFormData } from '@superset-ui/core';
+import Collapse from 'src/components/Collapse';
+
+type DataManager = {
+  formData: QueryFormData;
+  data: any[];
+  isInit: boolean;
+  isLoading: boolean;
+  isError: boolean;
+};
+
+// TODO: Fix this so links between dashboards will work again
+// const IP_DASHBOARD_ID = 19;
+// const IP_FILTER_ID = 'vxX3zR2Tz';
+// const SUPERSET_URL = 'http://localhost:9000';
+// const IP_DASHBOARD_LINK = `${SUPERSET_URL}/superset/dashboard/${IP_DASHBOARD_ID}/?native_filters=%28NATIVE_FILTER-${IP_FILTER_ID}%3A%28__cache%3A%28label%3APLEASEREPLACETHIS%2CvalidateStatus%3A%21f%2Cvalue%3A%21%28PLEASEREPLACETHIS%29%29%2CextraFormData%3A%28filters%3A%21%28%28col%3Aip_string%2Cop%3AIN%2Cval%3A%21%28PLEASEREPLACETHIS%29%29%29%29%2CfilterState%3A%28label%3APLEASEREPLACETHIS%2CvalidateStatus%3A%21f%2Cvalue%3A%21%28PLEASEREPLACETHIS%29%29%2Cid%3ANATIVE_FILTER-${IP_FILTER_ID}%2CownState%3A%28%29%29%29`;
+// END TODO
+
+/**
+ *   getPayloadField:
+ *     description: Returns the value for a given field.
+ *     parameters:
+ *       - name: field
+ *       - type: string
+ *       - required: true
+ *       - description: name of the field you want to get the value of
+ *
+ *       - name: payload
+ *       - type: any
+ *       - required: true
+ *       - description: data object containing the response from the server.
+ *     returns:
+ *       value:
+ *         description: Returns the value of the given field if found.
+ */
+const getPayloadField = (field: string, payload: any) => {
+  let value = '';
+  try {
+    if (payload !== undefined) {
+      value = payload[field];
+      if (value === null) {
+        value = 'Unknown';
+      }
+    }
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.log(e);
+    value = 'Something went wrong';
+  }
+  return value;
+};
+
+// Main Component
+function AtAGlanceUserIDCore(initialFormData: QueryFormData) {
+  const [userIDString, setUserIDString] = useState('coolguy@some.com,');
+
+  let canadianIpsList: any[] = [];
+  let nonCanadianIpsList: any[] = [];
+  let nonSucsessfulCanadianIpsList: any[] = [];
+  let nonSucsessfulNonCanadianIpsList: any[] = [];
```

Comment:
Would it make sense to call these "unsuccessful*IpsList"? (There's a typo in there as is.)

Reply:
I agree with you, I will get this changed, also I want to replace coolguy, I mean that's obviously Ryan's email address :)